packml_sm/src/packml_stats_provider.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2018 Plus One Robotics
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "packml_sm/packml_stats_provider.h"
+
+#include <limits>
+
+namespace packml_sm
+{
+PackmlStatsProvider::PackmlStatsProvider(std::shared_ptr<AbstractStateMachine> state_machine)
+  : state_machine_(state_machine)
+{
+}
+
+void PackmlStatsProvider::start()
+{
+  start_time_ = std::chrono::steady_clock::now();
```

Comment:
Does this time need to be stored locally, or can it be extracted from the state machine?

Reply:
I am not currently tracking it in the state machine, but I will